src/main/kotlin/ecommerce/controller/ProductController.kt

```diff
@@ -0,0 +1,70 @@
+package ecommerce.controller
+
+import ecommerce.model.Product
+import ecommerce.repository.ProductRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import java.net.URI
+
+@Controller
+@RequestMapping("/products")
+class ProductController {
+    @Autowired
+    private lateinit var jdbcTemplate: JdbcTemplate
```

Comment:
JdbcTemplate is better suited to be used only within the Repository layer rather than the Controller!